lib/Service/PageService.php

```diff
@@ -902,13 +817,89 @@ public function delete(int $collectiveId, int $parentId, int $id, string $userId
 		} catch (FilesNotPermittedException $e) {
 			throw new NotPermittedException($e->getMessage(), 0, $e);
 		}
-		$this->pageMapper->deleteByFileId($pageInfo->getId());
-		$this->removeFromSubpageOrder($collectiveId, $parentId, $id, $userId);
 
-		$this->revertSubFolders($folder);
+		$this->initTrashBackend();
+		if (!$this->trashBackend) {
+			// Delete directly if trash is not available
+			$this->pageMapper->deleteByFileId($id);
+			$this->removeFromSubpageOrder($collectiveId, $parentId, $id, $userId);
+			NodeHelper::revertSubFolders($folder);
+			// TODO: check
+			return $pageInfo;
+		}
+
+		$this->notifyPush($collectiveId, $userId);
+
+		$trashedPage = $this->pageMapper->findTrashedByFileId($id);
+		if (!$trashedPage) {
+			throw new NotFoundException('Failed to find trashed page in page trash: ' . $id);
```

Comment:
I cannot think of a precondition when this would happen. The check is mainly because in static analysis `$trashedPage` could be `null`. It should not happen in practice.

I agree with the problem to distinguish the cases. Therefore I changed the exception message to `Failed to find trashed page in page trash database`.

Reply:
I'm more referring to the status code. Parsing exception messages to destinguish server behaviors on the client side is always a bit of a mess. But yeah... we're probably not distinguishing them anyways. So could be done later in case we want to distinguish them.